pkg/controller/clusterpool/clusterpool_controller_test.go

```diff
@@ -944,26 +960,25 @@ func TestReconcileClusterPool(t *testing.T) {
 				assert.Equal(t, test.expectedObservedReady, pool.Status.Ready, "unexpected observed ready count")
 			}
 
-			missingDependentsCondition := controllerutils.FindClusterPoolCondition(pool.Status.Conditions, hivev1.ClusterPoolMissingDependenciesCondition)
-			if test.expectedMissingDependenciesStatus != nil {
-				require.NotNil(t, missingDependentsCondition)
-				expectedStatus := corev1.ConditionFalse
-				if *test.expectedMissingDependenciesStatus {
-					expectedStatus = corev1.ConditionTrue
+			if test.expectedMissingDependenciesStatus != "" {
+				missingDependenciesCondition := controllerutils.FindClusterPoolCondition(pool.Status.Conditions, hivev1.ClusterPoolMissingDependenciesCondition)
+				if assert.NotNil(t, missingDependenciesCondition, "did not find MissingDependencies condition") {
+					assert.Equal(t, test.expectedMissingDependenciesStatus, missingDependenciesCondition.Status,
+						"unexpected MissingDependencies conditon status")
+					if test.expectedMissingDependenciesMessage != "" {
```

Comment:
Is this check required? what happens if we don't have this check?

Reply:
Some of the tests have message present for the expected condition - I did not want to remove that additional check so I included it here, and at the same time, allow for just the reason to be compared.